src/pymor/vectorarrays/block.py

```diff
@@ -98,7 +98,7 @@ def dot(self, other):
         common_dtype = reduce(np.promote_types, (dot.dtype for dot in dots))
         ret = np.zeros(dots[0].shape, dtype=common_dtype)
         for dot in dots:
-            ret += dot
+            ret = ret + dot
```

Comment:
Is there any reason for this change?

Reply:
It seems I forgot to revert this when `promote_types` was added. I'll revert it now.